projects/asthma-copd/nbs/20_gene_corrs/output/aco/05-gene_expr_correlations-chr1.run.ipynb

```diff
@@ -0,0 +1,2336 @@
+{
```

Comment:
Up to you -- I don't have a strong opinion on this -- but do you think it's necessary to track all of these "output" scripts in your git repo? Usually when I do something like this with papermill I just add the output directory to my `.gitignore` file, and I only really look through all of the generated notebooks for each experiment if I need to debug some weird output or something.

Totally your decision, I generally try to keep the sizes of my git repos as small/manageable as I can, but that's just me.

Reply:
I was really looking for this kind of feedback here. Thank you, Jake!

I thought about this. My first idea was to save the "output" notebooks in the results directory, where the output files are generated (for instance, the gene-gene correlation matrices in this case). This makes sense because, to me, these notebooks are a more complex version of a "log file" (sometimes with tables, images, etc.). For example, the `05-gene_expr_correlation-chr...` notebooks have, for each chromosome, a plot with the distribution of correlations and a heatmap (like the one below). Initially, I was using this more as a debugging step, but it ended up being useful to have some of these plots available for the manuscript. It's also easy to visualize the notebooks in Github, whereas the results folder is never pushed to a repo. So these "output" notebooks are a mixture of a "log file" and a standard Jupyter notebook. Not sure about what to do, but I will think again with your comments in mind next time. Another alternative is to put the results directory under version control, which is something that I'm exploring right now with different tools beyond git.

Thank you again!

![image](https://user-images.githubusercontent.com/172687/208711211-535a4ab5-19d6-4127-8ae5-4fe0e35f6d22.png)